src/wma.ts

```diff
@@ -40,6 +40,25 @@ export class WMA {
      * does not affect any next calculations
      */
     momentValue(value: number) {
-        // return this.prevValue + (value - this.prevValue) / this.period;
+        const buffer = new CircularBuffer(this.buffer.length);
```

Comment:
please no :( 

mutate original buffer and then restore it, it synchronous operation and does not have any side effects

Reply:
push + pushback isn't work?